### ISSUE TITLE
shutdown via REST does not return HTTP response

### ISSUE THREAD

**Issue by justinc1** [cloudius-systems/osv#672]
/os/shutdown and poweroff (reboot too?) REST calls succeed, but caller doesn't get HTTP response back. VM is shut off "too fast". Even TCP RST is not sent, so caller just hangs waiting for data.

In effect you have to use really small timeout value for that calls.

**Comment by nyh** [comment]
You are right, this problem was already noted in the issue #624, but not solved there (because this was not the original point of that issue).

Issue #557 notes that one of the reasons why I wanted to support the SIOCOUTQ ioctl (see tcp(7) manual page) in OSv was so we can use it in the poweroff http request handling: My plan was for poweroff handler to *first* send the response, then wait until SIOCOUTQ becomes zero, meaning the remote end got the response, and then power off. Because we already have SIOCOUTQ working, we can implement this fix, and tt should be pretty easy to do it.

This will mean that the caller will get a success message a bit before the machine went down - but I think this is fine (the success message will mean that the machine is guaranteed to go down shortly), and the current situation is much worse because there is no way to even know if the vm got your poweroff request - you could be having network problems, and the vm never got your request.
